src/main/kotlin/snyk/container/annotator/ContainerYamlAnnotator.kt

```diff
@@ -13,27 +15,28 @@ import snyk.container.ContainerIssuesForImage
 class ContainerYamlAnnotator : ExternalAnnotator<PsiFile, Unit>() {
     private val logger = logger<ContainerYamlAnnotator>()
 
-    // overrides needed for the Annotator to invoke apply(). We don't do anything here
+    // override needed for the Annotator to invoke apply(). We don't do anything here
     override fun collectInformation(file: PsiFile): PsiFile? = file
-    override fun doAnnotate(collectedInfo: PsiFile?): Unit = Unit
 
-    fun getContainerIssuesForImages(psiFile: PsiFile): List<ContainerIssuesForImage> {
-        logger.debug("Calling doAnnotate on ${psiFile.name}")
+    // save all changes on disk to update caches through SnykBulkFileListener
+    override fun doAnnotate(collectedInfo: PsiFile?) {
+        logger.debug("doAnnotate on ${collectedInfo?.name}")
+        ApplicationManager.getApplication().invokeAndWait {
+            FileDocumentManager.getInstance().saveAllDocuments()
```

Comment:
I think this should only be done in the quickfix - annotating should not automatically save a document. But I'll be happy to add it to `AlwaysAvailableReplacementIntentionAction`

Reply:
Both quick-fixes and annotations based on info in our internal caches. And those caches are updated only on any file changes **on disk**. So to keep them up-to-date we need to flash all changes made by user in the editor (PSI) to the disk first. Otherwise we will have wrong annotations based on old file content shown in the editor:
![image](https://user-images.githubusercontent.com/14268320/152639312-ccf8a115-a4d0-4cd0-b15c-e08e7131d5f3.png)
![image](https://user-images.githubusercontent.com/14268320/152639362-bfd2c152-b859-4bd5-a03e-56e9da8c646e.png)